feat(DCS): add a new resource to manage big key analysis (#4030)
huaweicloud/terraform-provider-huaweicloud: docs/resources/dcs_bigkey_analysis.md

* `scan_type` - Indicates the analysis mode of the big key analysis. The value can be:
  + **manual**: indicates manual analysis.
  + **auto**: indicates automatic analysis.

* `created_at` - Indicates the creation time of the big key analysis. The format is **yyyy-mm-dd hh:mm:ss**.
  The value is in UTC format.

* `started_at` - Indicates the time when the big key analysis started. The format is **yyyy-mm-dd hh:mm:ss**.
  The value is in UTC format. (The value is null and is not returned when the analysis is being created.)

* `finished_at` - Indicates the time when the big key analysis ended. The format is **yyyy-mm-dd hh:mm:ss**.
  The value is in UTC format. (The value is null and is not returned when the analysis is being created.)

* `num` - Indicates the number of the big key.

* `status` - Indicates the analysis status. The value can be:
  + **waiting**: The analysis is waiting to begin.
  + **running**: The big key analysis is in progress.
  + **success**: The big key analysis succeeded.
  + **failed**: The big key analysis failed.

* `keys` - Indicates the record of big key. (The value is null and is not returned when the analysis is being created.)

>   The [keys](#DCS_keys) structure is documented below.

Reviewer: ```suggestion
  The [keys](#dcs_big_keys) structure is documented below.
```
DCS_keys is conflict with the hot key structure.

---
subcategory: "Distributed Cache Service (DCS)"
---

# huaweicloud_dcs_bigkey_analysis

Manage a DCS big key analysis resource within HuaweiCloud.

## Example Usage

```hcl
variable "instance_id" {}

resource "huaweicloud_dcs_bigkey_analysis" "test" {
  instance_id = var.instance_id
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.

* `instance_id` - (Required, String, ForceNew) Specifies the ID of the DCS instance.
  Changing this parameter will create a new resource.

## Attribute Reference

In addition to all arguments above, the following attribute is exported:

* `id` - The resource ID.

* `scan_type` - Indicates the analysis mode of the big key analysis. Valid value:
  + **manual**: indicates manual analysis.
  + **auto**: indicates automatic analysis.

* `created_at` - Indicates the creation time of the big key analysis. The format is yyyy-mm-dd hh:mm:ss.

>   The value is in UTC format. (The value is null and is not returned when the analysis is being created.)

Reviewer 2: ```suggestion
* `created_at` - Indicates the creation time of the big key analysis. The format is **yyyy-mm-dd hh:mm:ss**.
  The value is in UTC format.
```
and fix ohter places

* `started_at` - Indicates the time when the big key analysis started. The format is yyyy-mm-dd hh:mm:ss.
  The value is in UTC format. (The value is null and is not returned when the analysis is being created.)

* `finished_at` - Indicates the time when the big key analysis ended. The format is yyyy-mm-dd hh:mm:ss.
  The value is in UTC format.

* `num` - Indicates the number of the big key.

* `status` - Indicates the analysis status. Valid value:
  + **waiting**: The analysis is waiting to begin.
  + **running**: The big key analysis is in progress.
  + **success**: The big key analysis succeeded.
  + **failed**: The big key analysis failed.

* `keys` - Indicates the record of big key. (The value is null and is not returned when the analysis is being created.)
  The [keys](#DCS_keys) structure is documented below.

<a name="DCS_keys"></a>
The `keys` block supports:

* `name` - Indicates the name of big key.

* `type` - Indicates the type of big key. Valid value:
  + **string**
  + **list**
  + **set**
  + **zset**

>   + **hash**

Reviewer 2: ```suggestion
* `type` - Indicates the type of big key. The value can be **string**, **list**, **set**, **zset**, **hash**.
```

* `shard` - Indicates shard where the big key is located. This parameter is supported only when the instance type is cluster.

>   The format is ip:port.

Reviewer 2: ```suggestion
* `shard` - Indicates the shard where the big key is located. This parameter is supported only when the instance type is cluster.
  The format is **ip:port**.
```
